sql/core/src/test/resources/tpcds-plan-stability/approved-plans-v1_4/q92/explain.txt

```diff
@@ -1,5 +1,5 @@
 == Physical Plan ==
-TakeOrderedAndProject (34)
+* Sort (34)
```

Comment:
It changes from `TakeOrderedAndProject` to `Sort` seems because `Limit` after `Sort` is removed?

It might have additional shuffle for global `Sort`.

Reply:
This q92 sql:
```
SELECT sum(ws_ext_discount_amt) AS `Excess Discount Amount `
FROM web_sales, item, date_dim
WHERE i_manufact_id = 350
  AND i_item_sk = ws_item_sk
  AND d_date BETWEEN '2000-01-27' AND (cast('2000-01-27' AS DATE) + INTERVAL 90 days)
  AND d_date_sk = ws_sold_date_sk
  AND ws_ext_discount_amt >
  (
    SELECT 1.3 * avg(ws_ext_discount_amt)
    FROM web_sales, date_dim
    WHERE ws_item_sk = i_item_sk
      AND d_date BETWEEN '2000-01-27' AND (cast('2000-01-27' AS DATE) + INTERVAL 90 days)
      AND d_date_sk = ws_sold_date_sk
  )
ORDER BY sum(ws_ext_discount_amt)
LIMIT 100
```

yes, `Limit` after `Sort` is a special case, we will convert to `TakeOrderedAndProject`, but it seems not necessary to do both `sort` and `limit` if child maxRow == 1. Maybe we can do an another check seems like `if sort.child.maxRow <= 1 then remove sort` ?

Reply:
Other thought, we can add this pattern
```
case sort @ Sort(order, true, child)
          if sort.maxRow < conf.topKSortFallbackThreshold =>
        TakeOrderedAndProjectExec()
```

In this way, we can infer the `TakeOrderedAndProjectExec` from `Sort` which has not `Limit` after.

Reply:
Ah, @viirya , nice catch! Yea, how about simply excluding the case in `EliminateLimits`? @ulysses-you

Reply:
Updated.